controllers/bindings/dependents.go

```diff
@@ -32,20 +33,90 @@ type DependentsHandler struct {
 	Binding      *api.SPIAccessTokenBinding
 }
 
+// Dependents represent the secret and the list of the servicea accounts that are linked to a binding.
 type Dependents struct {
 	Secret          *corev1.Secret
 	ServiceAccounts []*corev1.ServiceAccount
 }
 
+type serviceAccountLink struct {
+	secret          bool
+	imagePullSecret bool
+}
+
+type serviceAccountNamesAndLinkTypes map[string]serviceAccountLink
+
+// CheckPoint is an opaque struct representing the state of the dependent objects at some point in time.
+// It can be used in the DependentsHandler.RevertTo method to delete the secret/service accounts from the cluster
+// that have been created after an instance of this struct has been returned from the DependentsHandler.CheckPoint
+// method.
+type CheckPoint struct {
+	secretName          string
+	serviceAccountNames serviceAccountNamesAndLinkTypes
+}
+
+func (s serviceAccountNamesAndLinkTypes) containsName(name string) bool {
+	for k := range s {
```

Comment:
I don't understand this. Is it checking that map contains some key? How is it different from 
```
if serviceAccount, contains := checkPoint.serviceAccountNames[sa.Name]; contains {
  ...
  if serviceAccount.secret {
    ...
  }
  ...
} else {
  ...
}
```

Reply:
Ah, yes, good catch.